components/recurring-contributions/UpdateOrderPopUp.js

```diff
@@ -82,30 +90,133 @@ const tiersQuery = gqlV2/* GraphQL */ `
+      amount: tier.amountType === 'FLEXIBLE' ? tier.minimumAmount.value * 100 : tier.amount.value * 100,
+      id: tier.id,
+      currency: tier.amount.currency,
+      interval: tier.interval,
+      presets: tier.presets,
+      minimumAmount: tier.amountType === 'FLEXIBLE' ? tier.minimumAmount.value * 100 : 100,
+    }));
+  if (!disableCustomContributions) {
+    tierOptions.unshift({
+      key: `${contribution.id}-custom-tier`,
+      title: intl.formatMessage(messages.customTier),
+      flexible: true,
+      amount: 100,
+      id: null,
+      currency: contribution.amount.currency,
+      interval: contribution.frequency.toLowerCase().slice(0, -2),
+      presets: [500, 1000, 2000, 5000],
+      minimumAmount: 100,
+      isCustom: true,
+    });
   }
-  // selectedTier.presets if it's the default tier, but selectedTier.value.preset afterwards if it's a radio list selection
-  const presets = selectedTier.presets || selectedTier.value?.presets || [500, 1000, 2000, 5000];
+  return tierOptions;
+};
 
-  presets.map(preset => {
-    optionObject = {
-      label: formatCurrency(preset, contribution.amount.currency),
-      value: preset,
-    };
-    objectArray.push(optionObject);
-  });
-  objectArray.push({ label: 'Other', value: 100 });
-  return objectArray;
+const getDefaultContributeOption = (contribution, tiersOptions) => {
+  const customContribution = tiersOptions.find(option => option.isCustom);
+  if (!contribution.tier) {
+    return customContribution;
+  } else {
+    // for some collectives if a tier has been deleted it won't have moved the contribution
+    // to the custom 'null' tier so we have to check for that
+    const matchedTierOption = tiersOptions.find(option => option.id === contribution.tier.id);
+    return !matchedTierOption ? customContribution : matchedTierOption;
+  }
+};
+
+const useContributeOptions = (order, tiers, disableCustomContributions) => {
+  const intl = useIntl();
+  const [loadingDefaultTier, setLoadingDefaultTier] = useState(true);
+  const [selectedContributeOption, setSelectedContributeOption] = useState(null);
+  const [amountOptions, setAmountOptions] = useState(null);
+  const [selectedAmountOption, setSelectedAmountOption] = useState(null);
+  const contributeOptions = React.useMemo(() => {
+    return getContributeOptions(intl, order, tiers, disableCustomContributions);
+  }, [intl, order, tiers, disableCustomContributions]);
+
+  useEffect(() => {
+    if (contributeOptions && !selectedContributeOption) {
+      setSelectedContributeOption(getDefaultContributeOption(order, contributeOptions));
+      setLoadingDefaultTier(false);
+    }
+  }, [contributeOptions]);
+
+  useEffect(() => {
+    if (selectedContributeOption !== null) {
+      const options = getTierAmountOptions(selectedContributeOption?.value, order);
+      setAmountOptions(options);
+      setSelectedAmountOption(first(options));
+    }
+  }, [selectedContributeOption]);
+
+  return {
+    loadingDefaultTier,
```

Comment:
Since this is the only loading state of this hook, I would rename it for consistency:
```suggestion
    loading,
```

Down the code you can also update to use
```
contributeOptionsState.loading
```

Reply:
That's right :+1: